File: acmicpc/1260/1260.py
import queue

n, m, v = [ int(i) for i in input().split() ]
if n == m == v == 1:
    print(1)
    print(1)
    exit(0)
g = [ [] for _ in range(n) ]
for _ in range(m):
    a, b = [ int(i) for i in input().split() ]
    g[a-1].append(b-1) 
    g[b-1].append(a-1)

for i in g:
    i.sort()

def bfs(g, v):
    result = []
    q = queue.Queue()
    result.append(v-1)
    q.put(v-1)

    while not q.empty() :
        item = q.get()
        for j in g[item]:
            if j not in result:
                result.append(j)
                q.put(j)
    
    return result


def dfs(g, v, result):
    result.append(v)

    for i in g[v]:
        if i not in result:
            dfs(g, i, result)
    
result = []
dfs(g,v-1,result)
print([ i+1 for i in result ])
print([ i+1 for i in bfs(g,v) ])

File: acmicpc/1260/1260.cc
#include <cstdio>
#include <cstring>
#include <queue>
using namespace std;
int n, m, start;
int tree[1001][1001] = {0,};
int visit[1001] = {0,};
void dfs(int v) {
    visit[v] = 1;
    printf("%d ", v);
    for (int i = 1; i <= 1000; i++) {
        if (tree[v][i] == 1 && visit[i] != 1) {
            dfs(i);
        }
    }
}

void bfs(int v) {
    queue<int> q;
    q.push(v);
    visit[v] = 1;

    while (!q.empty()) {
        int c = q.front(); q.pop();
        printf("%d ", c);
        for (int i = 1; i <= 1000; i++) {
            if (tree[c][i] == 1 && visit[i] != 1) {
                q.push(i);
                visit[i] = 1;
            }
        }
    }
}
int main() {
    scanf("%d %d %d", &n, &m, &start);
    for (int i = 0; i < m; i++) {
        int a, b; scanf("%d %d", &a, &b);
        tree[a][b] = 1;
        tree[b][a] = 1;
    }
    dfs(start); printf("\n");
    memset(visit, 0, sizeof(visit));
    bfs(start);
}

